feature/619

AnimeOneLove2190 opened the pull request:
Methods have been moved to the correct namespaces
Old namespaces have not yet been deleted

AnimeOneLove2190 committed:
feature/619
Methods have been moved to the correct namespaces
Old namespaces have not yet been deleted

AnimeOneLove2190 committed:
feature/619/2
Fixed GetSeansViewList method

AnimeOneLove2190 committed:
feature/619/3
Fixed GetTicketViewList method

AnimeOneLove2190 committed:
feature/619/4
Add Loggers in SesssionService and TicketService
SatisticService and StatisticsController removed

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/SessionService.cs:
```diff
@@ -214,14 +252,41 @@ public async Task DeleteAsync(int id)
+        {
+            var startDate = start ?? DateTime.Today;
+            var endDate = end ?? DateTime.Today;
+            var sessionsInPeriod = await sessionDataAccess.GetSessionListInPeriodAsync((DateTime)start, (DateTime)end);
```
Параметры startDate и endDate

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/SessionService.cs:
```diff
@@ -18,57 +21,71 @@ public class SessionService : ISessionService
+                logger.LogError(exceptionMessage);
+                throw new CustomException(exceptionMessage);
             }
             var sessionWithTheSameStart = await sessionDataAccess.GetSessionAsync(sessionCreate.Start);
```
Получить сеанс и по времени и по холлу. Перед этим проверить существования зала.
В текущем кейсе ты можешь получить сеанс который совпадает по времени но находится в другом зале

artemlipinskiy commented on CinemaTicket.WebApi/Controllers/TicketController.cs:
```diff
@@ -48,5 +48,23 @@ public async Task DeleteTicketAsync(int id)
         }
+        [HttpGet]
+        [Route("GetTicketViewList")]
+        public async Task<List<TicketView>> GetTicketViewListAsync(int sessionId, Nullable<bool> isSold)
```
(int sessionId, bool? isSold)

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/TicketService.cs:
```diff
@@ -150,13 +194,125 @@ public async Task DeleteAsync(int id)
+                throw new NotFoundException(exceptionMessage);
+            }
+            var unsoldTickets = ticketsFromDB.Where(x => x.IsSold == false).ToList();
+            if (unsoldTickets == null)
```
Невозможна ситуация, что unsoldTickets == null, т.к. получается путем ToList - будет пустой список.
Если хочешь убедиться, что нет проданных билетов, сравни кол-во, которое запросили и количество IsSold == false

AnimeOneLove2190 replied:
Тут я скорее хотел убедиться, что все запрошенные билеты уже проданы

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/TicketService.cs:
```diff
@@ -150,13 +194,125 @@ public async Task DeleteAsync(int id)
+                logger.LogError(exceptionMessage);
+                throw new NotFoundException(exceptionMessage);
+            }
+            var unsoldTickets = ticketsFromDB.Where(x => x.IsSold == false).ToList();
```
Аналогичная ситуация. Можно проверить есть ли проданные билеты. т.е. кол-во x.IsSold == true > 0

AnimeOneLove2190 replied:
Аналогично здесь, если все запрошенные билеты проданы, то удалять ничего нельзя, если часть из запрошенных проданные, то их не трогаем, а удаляем только непроданные

AnimeOneLove2190 committed:
feature/619/5
Changes have been made, but not all; remaining comments require further conversation

artemlipinskiy commented on CinemaTicket.Infrastructure/Constants/ExceptionMessageTemplate.cs:
```diff
@@ -13,8 +13,10 @@ public static class ExceptionMessageTemplate
         public const string Duplicate = "{0} list has one or more duplicate.";
         public const string NotAllFound = "Not all items from the {0} were found in the database";
+        public const string TicketIsSold = "Ticket with Id {0} has already been sold, it cannot be changed";
+        public const string UnhandledException = "I don't know how to properly describe this exception";
```
One or more tickets are sold

AnimeOneLove2190 committed:
feature/619/6
The exception pattern has been renamed
